src/shogun/lib/SGMatrix.h

```diff
@@ -26,6 +28,28 @@ namespace Eigen
 	template <class, int, class> class Map;
 }
 
+/**
+ * Useful type traits for template overloads
+ */
+// Checks if any one of the types matches
+// Ref: https://stackoverflow.com/a/17032517/3656081
+template <typename T, typename... Rest>
+struct is_any : std::false_type
```

Comment:
I wonder whether this is the best place to put this?

Reply:
It might be worth having a file with all the utility type_traits, no? Something like `sg_type_traits.h`

Reply:
Yep!